### PR TITLE
feat(replays): Add urls, user_agent, and replay_start_timestamp migration

### PR DESCRIPTION
**Pull request by cmanallen**
Companion of: https://github.com/getsentry/snuba/pull/3023

- Adds migration.

### CODE REVIEW

**Comment by cmanallen** [comment]
Updated because of broken migration.

This PR has a migration; here is the generated SQL

```sql
-- start migrations

-- forward migration replays : 0005_add_urls_user_agent_replay_start_timestamp
Local operations:
ALTER TABLE replays_local ADD COLUMN IF NOT EXISTS urls Array(String) AFTER url;
ALTER TABLE replays_local ADD COLUMN IF NOT EXISTS replay_start_timestamp Nullable(DateTime) AFTER timestamp;
ALTER TABLE replays_local ADD COLUMN IF NOT EXISTS user_agent Nullable(String) AFTER user;
ALTER TABLE replays_local DROP COLUMN IF EXISTS url;


Dist operations:
ALTER TABLE replays_dist ADD COLUMN IF NOT EXISTS urls Array(String) AFTER url;
ALTER TABLE replays_dist ADD COLUMN IF NOT EXISTS replay_start_timestamp Nullable(DateTime) AFTER timestamp;
ALTER TABLE replays_dist ADD COLUMN IF NOT EXISTS user_agent Nullable(String) AFTER user;
ALTER TABLE replays_dist DROP COLUMN IF EXISTS url;
-- end forward migration replays : 0005_add_urls_user_agent_replay_start_timestamp




-- backward migration replays : 0005_add_urls_user_agent_replay_start_timestamp
Local operations:
ALTER TABLE replays_local DROP COLUMN IF EXISTS user_agent;
ALTER TABLE replays_local DROP COLUMN IF EXISTS replay_start_timestamp;
ALTER TABLE replays_local DROP COLUMN IF EXISTS urls;
ALTER TABLE replays_local ADD COLUMN IF NOT EXISTS url Nullable(String) AFTER title;


Dist operations:
ALTER TABLE replays_dist DROP COLUMN IF EXISTS user_agent;
ALTER TABLE replays_dist DROP COLUMN IF EXISTS replay_start_timestamp;
ALTER TABLE replays_dist DROP COLUMN IF EXISTS urls;
ALTER TABLE replays_dist ADD COLUMN IF NOT EXISTS url Nullable(String) AFTER title;
-- end backward migration replays : 0005_add_urls_user_agent_replay_start_timestamp
```

File: snuba/migrations/snuba_migrations/replays/0005_add_urls_user_agent_replay_start_timestamp.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
+from typing import List, Sequence, Tuple
+
+from snuba.clickhouse.columns import Array, Column, DateTime, String
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+
+new_columns: Sequence[Tuple[Column[Modifiers], str]] = [
+    (Column("urls", Array(String())), "url"),
```
Comment (evanh) [inline snuba/migrations/snuba_migrations/replays/0005_add_urls_user_agent_replay_start_timestamp.py:9]: I don't think this really matters, but I wouldn't make the "after" a column that is being removed.

File: snuba/datasets/replays_processor.py [hunk 2]
```diff
@@ -66,10 +65,13 @@ def _process_base_replay_event_values(
         processed["timestamp"] = self.__extract_timestamp(
             replay_event["timestamp"],
         )
+        processed["replay_start_timestamp"] = self.__extract_started_at(
+            replay_event.get("replay_start_timestamp"),
+        )
+        processed["urls"] = replay_event.get("urls", [])
```
Comment (evanh) [inline snuba/datasets/replays_processor.py:71]: Is there a limit on how many URLs can be in this list?

**Review by evanh — COMMENTED** [review]
Please see comments.

**Comment by cmanallen** [comment]
@evanh Updated with the processors removed.  The tests fail.  Is that okay?

**Comment by cmanallen** [comment]
@evanh Split the branch and made the changes backwards compatible.

**Comment by evanh** [comment]
Hey, I noticed you removed the delete column command, which was smart (that should be a third PR that happens after the processor PR). However now your backwards steps are not correct. You're adding the `url` column back which was never removed.

**Comment by cmanallen** [comment]
@evanh Thanks good catch.  I've added some new fields and removed the user_agent field.  Re: https://www.notion.so/sentry/Handling-of-new-client-hints-to-parse-user-agent-information-3590fcbbdbd542faa6fa07ba67c0ddb3

I'm using a limited subset of the `contexts` object's fields.

**Review by onewland — APPROVED** [review]

**Pull request merged**